ornate_frontend/src/pages/Home.js

```diff
@@ -0,0 +1,302 @@
+        <div className="cards">
+          {/* Image */}
+          <img src={card} alt="card" className="display_class" />
+          {/* overlay on hovering */}
+          <div className="overlay_image">
+            <div className="user_actions">
+              <div id="like_section">
+                <button className="like_button">
+                  <FcLike className="like" />
+                  {/* <FcLikePlaceholder className="like" /> */}
+                </button>
+              </div>
+
+              <div id="save_section">
+                <button className="save_button">
+                  Save <CiSaveDown2 />
+                </button>
+              </div>
+            </div>
+          </div>
+        </div>
+
+        <div className="cards">
+          {/* Image */}
+          <img src={card} alt="card" className="display_class" />
+          {/* overlay on hovering */}
+          <div className="overlay_image">
+            <div className="user_actions">
+              <div id="like_section">
+                <button className="like_button">
+                  <FcLike className="like" />
+                  {/* <FcLikePlaceholder className="like" /> */}
+                </button>
+              </div>
+
+              <div id="save_section">
+                <button className="save_button">
+                  Save <CiSaveDown2 />
+                </button>
+              </div>
+            </div>
+          </div>
+        </div>
+
+        <div className="cards">
+          {/* Image */}
+          <img src={card} alt="card" className="display_class" />
+          {/* overlay on hovering */}
+          <div className="overlay_image">
+            <div className="user_actions">
+              <div id="like_section">
+                <button className="like_button">
+                  <FcLike className="like" />
+                  {/* <FcLikePlaceholder className="like" /> */}
+                </button>
+              </div>
+
+              <div id="save_section">
+                <button className="save_button">
+                  Save <CiSaveDown2 />
+                </button>
+              </div>
+            </div>
+          </div>
+        </div>
+
+        <div className="cards">
+          {/* Image */}
+          <img src={card} alt="card" className="display_class" />
+          {/* overlay on hovering */}
+          <div className="overlay_image">
+            <div className="user_actions">
+              <div id="like_section">
+                <button className="like_button">
+                  <FcLike className="like" />
+                  {/* <FcLikePlaceholder className="like" /> */}
+                </button>
+              </div>
+
+              <div id="save_section">
+                <button className="save_button">
+                  Save <CiSaveDown2 />
+                </button>
+              </div>
+            </div>
+          </div>
+        </div>
+
+        <div className="cards">
```

Comment:
Small Code optimization can be achieved by extracting the repeated code into a new component and implementing loops to efficiently execute it as many times as required.